### PR TITLE
Workload identity federation example

### PR DESCRIPTION
**Pull request by apichick**

### CODE REVIEW

File: examples/cloud-operations/workload-identity-federation/azure.tf [hunk 1]
```diff
@@ -0,0 +1,177 @@
+locals {
+  username    = "azureuser"
+  app_name    = "test-app"
+  custom_data = <<CUSTOM_DATA
+#!/bin/bash
+cat <<EOF > /home/${local.username}/credentials.json
+{
```
Comment (ludoo) [inline examples/cloud-operations/workload-identity-federation/azure.tf:7]: I would put this into an external file and then call `templatefile()`

File: examples/cloud-operations/workload-identity-federation/google-cloud.tf [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  source = "../../../modules/project"
+  name   = var.project_id
+  services = [
+    "cloudresourcemanager.googleapis.com",
+    "iam.googleapis.com",
+    "iamcredentials.googleapis.com",
+    "sts.googleapis.com",
+  ]
+  project_create = var.project_create
+  iam = {
```
Comment (ludoo) [inline examples/cloud-operations/workload-identity-federation/google-cloud.tf:27]: you need to use `iam_additive` with an existing project, or you'll nuke any pre-existing viewer

File: examples/cloud-operations/workload-identity-federation/variables.tf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_create" {
+  description = "Flag indicating whether the Google Cloud projects should be created"
```
Comment (ludoo) [inline examples/cloud-operations/workload-identity-federation/variables.tf:18]: trailing dot (`.`) here and in the next variable

File: examples/cloud-operations/workload-identity-federation/azure.tf [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+data "azuread_client_config" "config" {}
+
+resource "azuread_application" "app" {
```
Comment (ludoo) [inline examples/cloud-operations/workload-identity-federation/azure.tf:51]: As discussed, I would make Azure resource creation (vm, etc) optional if users have existing resources they want to leverage.

**Comment by ludoo** [comment]
Can you also add a simple test?

File: examples/cloud-operations/workload-identity-federation/azure.tf [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+EOF
+sudo apt -y update
+sudo apt -y install apt-transport-https ca-certificates gnupg jq
+echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+sudo apt-get update && sudo apt-get install google-cloud-sdk
+CUSTOM_DATA
+}
+
+/**
```
[the remark this answers is not in the document]
Comment (apichick) [reply examples/cloud-operations/workload-identity-federation/azure.tf:1]: done

File: examples/cloud-operations/workload-identity-federation/azure.tf (hunk 1 above)
Comment (apichick) [reply examples/cloud-operations/workload-identity-federation/azure.tf]: done

File: examples/cloud-operations/workload-identity-federation/google-cloud.tf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    issuer_uri        = "https://sts.windows.net/${data.azuread_client_config.config.tenant_id}"
+  }
+}
+
+module "sa" {
+  source     = "../../../modules/iam-service-account"
+  project_id = module.prj.project_id
+  name       = "sa-test"
+  iam = {
+    "roles/iam.workloadIdentityUser" : ["principalSet://iam.googleapis.com/projects/${module.prj.number}/locations/global/workloadIdentityPools/${google_iam_workload_identity_pool.pool.workload_identity_pool_id}/*"]
```
[the remark this answers is not in the document]
Comment (apichick) [reply examples/cloud-operations/workload-identity-federation/google-cloud.tf]: done

File: examples/cloud-operations/workload-identity-federation/azure.tf (hunk 4 above)
Comment (apichick) [reply examples/cloud-operations/workload-identity-federation/azure.tf:28]: done

**Review by ludoo — APPROVED** [review]

**Comment by apichick** [comment]
Added test

**Pull request merged**